test/es-module/test-esm-loader-entry-url.mjs

```diff
@@ -0,0 +1,99 @@
+import { spawnPromisified } from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import assert from 'node:assert';
+import { execPath } from 'node:process';
+import { describe, it } from 'node:test';
+
+// Helper function to assert the spawned process
+async function assertSpawnedProcess(args, options = {}, expected = {}) {
+  const { code, signal, stderr, stdout } = await spawnPromisified(execPath, args, options);
+
+  if (expected.stderr) {
+    assert.match(stderr, expected.stderr);
+  }
+
+  if (expected.stdout) {
+    assert.match(stdout, expected.stdout);
+  }
+
+  assert.strictEqual(code, expected.code ?? 0);
+  assert.strictEqual(signal, expected.signal ?? null);
+}
+
+// Common expectation for experimental feature warning in stderr
+const experimentalFeatureWarning = { stderr: /--entry-url is an experimental feature/ };
+
+describe('--entry-url', { concurrency: true }, () => {
+  const doubleEncodingPath = fixtures.fileURL('es-modules/test-esm-double-encoding-native%20.mjs').pathname;
+
+  it('should reject loading absolute path that contains %', async () => {
+    await assertSpawnedProcess(
+      ['--entry-url', decodeURIComponent(doubleEncodingPath)],
```

Comment:
Minor nit: after the fix, the test title became misleading: the decoded path doesn't contain `%`.
Perhaps something like `it should reject loading path decoded from URL ` would be more correct here.

Reply:
Or maybe pass a path instead, so the test description matches.

Reply:
The only reason it seems misleading is that on windows, "C:/" is treated as a protocol, and not a path. Because of that, I used this method to pass a path, as it ensures the path begins with "/". 

Because of that, I'm going to elect to ignore this nitpick, however, if you feel strongly about it, and you have a better method, I'm happy to apply.

Reply:
```suggestion
  const doubleEncodingPath = fixtures.fileURL('es-modules/test-esm-double-encoding-native%20.mjs').pathname;

  it('should reject loading absolute path that contains %', async () => {
    await assertSpawnedProcess(
      ['--entry-url', './test-esm-double-encoding-native%20.mjs'],
      { cwd: path.fileURL('es-modules') },
```